src/bitdrift_public/protobuf/client/v1/api.proto

```diff
@@ -197,9 +197,33 @@ message ApiRequest {
     ConfigurationUpdateAck runtime_update_ack = 5;
     OpaqueConfigurationUpdateAck opaque_configuration_update_ack = 8;
     OpaqueRequest opaque_upload = 9;
+    SankeyDiagramUploadRequest sankey_diagram_upload = 10;
   }
 }
 
+// A request to upload a Sankey diagram.
+message SankeyDiagramUploadRequest {
+  // A single node in the Sankey diagram. This differs from workflow states. Each node corresponds
+  // to a single transition from a workflow origin state to a workflow target state.
+  message Node {
+    // The ID of the origin state.
+    string origin_workflow_state_id = 1 [(validate.rules).string = {min_len: 1}];
```

Comment:
would this be the generated client ID? Or the FE ID? Not sure how useful the generated ID would be since I it might vary between workflow constructions (I think)